src/components/UpcomingEventsCard/index.jsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import './style.css';
+import bg2 from './img/bg_2.png';
+import { charityAPI } from '../../clients';
+
+class UpcomingEventsCard extends React.Component {
+  state = {
+    date: {},
+    loading: true,
+    error: false,
+    errorMessage: ''
+  };
+
+  _getData = () => {
+    this.setState({ loading: true });
+    charityAPI('/featured-cause')
+      .then(({ data }) => {
+        this.setState({
+          data: data,
+          loading: false,
+          error: false,
+          errorMessage: ''
+        });
+      })
+      .catch(error => {
+        this.setState({
+          loading: false,
+          error: true,
+          errorMessage: "couldn't fetch data"
+        });
+      });
+  };
+
+  componentDidMount() {
+    this._getData();
+  }
+
+  progress = (raised, goal) => {
+    return Math.floor((raised / goal) * 100);
+  };
+
+  render() {
+    if (this.state.loading) {
+      return <div>Loading...</div>;
+    }
+
+    if (this.state.error) {
+      return (
+        <div>
+          {this.state.errorMessage},{' '}
+          <a href="#/" onClick={this._getData} className="text-c200">
+            retry?
+          </a>
+        </div>
+      );
+    }
+
+    return (
+      <div className=" Upcoming-Events-Card w-3/5 relative z-20">
+        <div className="absolute z-30">
+          <img src={bg2} alt="upcoming events right-bg" />
+        </div>
+        <div className="h-full z-40 right-fund-card text-c000 flex flex-col items-center justify-evenly">
+          <div className="guage">
+            <div className="percentage-circle">
+              <div className="percent">
+                <svg className="main_circle">
+                  <circle cx="100" cy="100" r="90"></circle>
+                  <circle
+                    cx="100"
+                    cy="100"
+                    r="90"
+                    style={{
+                      strokeDashoffset: `${
+                        this.progress(
+                          this.state.data.featured_cause.raised,
+                          this.state.data.featured_cause.Goal
+                        ) * -5.65
+                      }`
+                    }}
+                  ></circle>
+                </svg>
+                <div className="number">
+                  <h2>
+                    {this.progress(
+                      this.state.data.featured_cause.raised,
+                      this.state.data.featured_cause.Goal
+                    )}
+                    %
+                  </h2>
+                </div>
+                {/* <svg className="dot">
```

Comment:
is this suppose to be the circle in the design?

Reply:
Yup .. but I couldn't figure how to make it dynamic with the progress bar.

Reply:
**_Yes_** this is the small circle. 
We had a challenge in how we make this circle moves when the percentage value of progress changes.
![Capture](https://user-images.githubusercontent.com/8650435/80743014-55f30300-8b2d-11ea-802c-f5261857f6b3.PNG)

**Our approach**
-  
We can make the position of small circle `(top` and `left` property) related to the percentage
of progress value.

**Or**

We can make it fixed.